src/layer.js

```diff
@@ -145,50 +145,63 @@ export default class Layer {
    * @return {Promise} [will resolve once image src has been retrieved and dimensions calculated]
    */
   calculateSize() {
```

Comment:
It might be worth moving this logic into it's own method as it is not relating to size calculation, also needs a bit of documentation :)